### PR TITLE
Documentation for MyBlocks

### PR DESCRIPTION
**Pull request by koonsies**
In an effort to flesh out the documents, I wrote a bit about MyBlocks/Cross Site Verify/Secure Blocks. A good amount of this is just my current impression of the tools. Please review and let me know of any corrections that should be made.

### CODE REVIEW

File: myblocks/models.py [hunk 1]
```diff
@@ -811,12 +811,13 @@ def handle_job_detail_redirect(self, request, *args, **kwargs):
         """
         Used on job detail pages, this returns a redirect to the
         appropriate url if:
-            1. There is no matching job (404).
-            2. The matching job isn't actually available on the site
-               because it belongs to a business unit or site package
-               not associated with that site (Redirect to home page).
-            3. It's missing slugs or the slugs are out of order
-               (Redirect to the page with slugs in the correct order).
+
```
Comment (koonsies) [inline myblocks/models.py]: I made this correction because Sphinx was choking on the indents here.

**Comment by koonsies** [comment]
By the way, great job on getting this set up @aspidites .. this is a much more elegant way to handle documentation.

File: doc/source/conf.py [hunk 2]
```diff
@@ -20,8 +20,6 @@
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 sys.path.insert(0, os.path.abspath('../..'))
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'settings')
-print os.path.abspath(__file__)
```
Comment (aspidites) [inline doc/source/conf.py]: Dang. I didn't realize I had left this here. Thanks.

File: doc/source/developers/myblocks/index.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
+===============
+MyBlocks
+===============
+MyBlocks is a designed as a platform for creating highly customizable, modular components called Blocks. Blocks can be
+used independently on a page, or together to form an entire webpage.
+
+Overview
```
Comment (aspidites) [inline doc/source/developers/myblocks/index.rst]: Stylistic question. I had been moving `Overview` sections to index.rst and putting anything else in files that seemed relevant (see postajob for instance). The question is, does it make more sense to have a short blurb and a dedicated overview section as you've done here, or continue how I have been doing. Either way, I'd like us to remain consistent.

File: doc/source/developers/myblocks/index.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+===============
+
+MyBlocks was originally conceived as an idea to extend the customization of Microsites webpages while offering a method
+for ease of reuse of components. With MyBlocks, each block represents a bootstrap div with spans and offsets to allow
+sizing of each module. Nesting of blocks can also occur, for instance: a body block may job listing block, which contains
```
Comment (aspidites) [inline doc/source/developers/myblocks/index.rst]: If there are blocks that correspond to these "job listing blocks", It'd be nice to note that. Something like:

> ```myblocks.models.JobListingBlock```

If it exsists.

File: doc/source/developers/index.rst [hunk 5]
```diff
@@ -5,6 +5,7 @@ Developer Documentation
 .. toctree::
   :maxdepth: 1
 
+  myblocks/index.rst
```
Comment (aspidites) [inline doc/source/developers/index.rst]: .rst is redundant here.

File: doc/source/developers/myblocks/index.rst (hunk 3 above)
Comment (koonsies) [inline doc/source/developers/myblocks/index.rst]: Good question. My personal preference tends to be to have a good detailing of a module before I start going into any navigation.  On each module, do we want an extended overview in addition to a TLDR-esque blurb, or just one succinct description for module documentation main pages.

@darrint @fezick @JLMcLaugh @tdphillips @lemniscate613 opinions, guys?

File: doc/source/developers/myblocks/index.rst (hunk 4 above)
Comment (koonsies) [inline doc/source/developers/myblocks/index.rst]: Hmm. I don't have any examples off the top of my head. This was written purely as my interpretation of MyBlocks, so hopefully anyone with more knowledge could help me flesh that out or provide better examples.

File: doc/source/developers/myblocks/index.rst (hunk 3 above)
Comment (aspidites) [inline doc/source/developers/myblocks/index.rst]: My only vote is that whichever we do, we do consistently. I prefer succinct, but wouldn't argue either way so long as we are consistent.

File: doc/source/developers/myblocks/index.rst (hunk 4 above)
Comment (aspidites) [inline doc/source/developers/myblocks/index.rst]: How about Login Blocks, or the soon to be saved search list block? If you go to /admin and scroll to the MyBlocks section, there are a few other examples.

File: doc/source/developers/myblocks/index.rst [hunk 6]
```diff
@@ -0,0 +1,10 @@
+=========
+MyBlocks
+=========
+MyBlocks is a designed as a platform for creating highly customizable, modular
```
Comment (aspidites) [inline doc/source/developers/myblocks/index.rst]: Small, I know, but missing space between the `=`s and the sentence.

**Comment by aspidites** [comment]
Docstrings are inconsistent. For instance, I see both:

```
"""
     This is a test
     of multiline docstring
"""
```

and

""" this is a test
    of multiline docstring
"""

I think elsewhere we've been using:

"""
this is a test
of multiline docstring
"""

Once that + whitespace are resolved, I'm ready to merge this, unless someone else objects in the mean time.

**Comment by koonsies** [comment]
Do we have a standard for how the docstrings should be written? My inconsistency comes from having not yet chosen a preferred way, though I tend to like the first.

**Comment by aspidites** [comment]
@koonsies https://github.com/DirectEmployers/private-docs/blob/master/developer/workflow_style_guide.rst

**Comment by aspidites** [comment]
Same failure I had before. I think it's transient. Not sure though.

**Pull request merged**